### PR TITLE
Proposed fix for UP flag presence

### PR DESCRIPTION
**Pull request by efabris**
Both in the Assertion and Attestation validation the WebAuthn guideline specify these controls:

- Verify that the User Present bit of the flags in authData is set.
- If user verification is required for this registration, verify that the User Verified bit of the flags in authData is set.

Despite this recommendation, the library check the UP flag only if "second" factor is specified in the expectations. I reported my observations in [this issue](https://github.com/webauthn-open-source/fido2-lib/issues/57) and now I'm proposing a fix.

### CODE REVIEW

File: test/validatorTest.js [hunk 1]
```diff
@@ -720,6 +720,24 @@ describe("attestation validation", function() {
 					return assert.isRejected(attResp.validateFlags(), Error, "expected flag was not set: ED");
 				});
 
+				it("throws if UV and UV is set but not UP", function() {
```
Comment (JamesCullum) [inline test/validatorTest.js:723]: If UV and UV is set? Is that a typo?

File: test/validatorTest.js [hunk 2]
```diff
@@ -720,6 +720,24 @@ describe("attestation validation", function() {
...
+					attResp.expectations.set("flags", ["UV"]);
+					attResp.authnrData.set("flags", new Set(["UV"]));
+					return assert.isRejected(attResp.validateFlags(), Error, "expected User Presence (UP) flag to be set if User Verification (UV) is set");
+				});
+
+				it("throws if UV but UV is not set", function() {
```
Comment (JamesCullum) [inline test/validatorTest.js:729]: Same here - I think the wording can be a bit more concise

**Review by JamesCullum — COMMENTED** [review]
Thanks a lot for your great addition. If you could make the test names more easily understandable, that'd be great.

**Comment by JamesCullum** [comment]
Also it would be great if you could already bump the version, so that it can be published on merge 👍

File: test/validatorTest.js (hunk 1 above)
Comment (efabris) [reply test/validatorTest.js:723]: I have used the style already present in the test, for example [here](https://github.com/webauthn-open-source/fido2-lib/blob/993521509af7b6e25eb40513a417b7fa31ddacce/test/validatorTest.js#L739): "_throws if UP-or-UV and neither is set_".

Maybe, using the same style, I can write: "_throws if UV and UP is not set_".

Do you have any suggestions to improve the test name?

File: test/validatorTest.js (hunk 2 above)
Comment (efabris) [reply test/validatorTest.js:729]: Same here, maybe I can write: "_throws if UV and UV is not set_".

Do you have any suggestions to improve the test name?

**Review by efabris — COMMENTED** [review]
I added some on your hint, @JamesCullum.

File: test/validatorTest.js (hunk 1 above)
Comment (JamesCullum) [reply test/validatorTest.js:723]: In this case I believe "throws if UV is set but UP is not set" - same for the other one

File: test/validatorTest.js (hunk 2 above)
Comment (JamesCullum) [reply test/validatorTest.js:729]: "throws if UV is not set"? Does it always HAVE to be set, by the way?

Comment (efabris) [reply test/validatorTest.js]: No, UV is not mandatory. But if UV is present UP must also be present, as per webauthn specification both for [registration (14th element in the check list)](https://www.w3.org/TR/webauthn-2/#sctn-registering-a-new-credential) and for [authentication (16th element in the check list)](https://www.w3.org/TR/webauthn-2/#sctn-verifying-assertion).

**Comment by JamesCullum** [comment]
Thanks a lot for your great contributions and high quality!

**Pull request merged**